fl4health/preprocessing/warmed_up_module.py

```diff
@@ -0,0 +1,121 @@
+                log(INFO, f"Weights mapping dict: {self.weights_mapping_dict}")
+        else:
+            log(INFO, "Weights mapping dict is not provided. Matching stats directlly, based on current model's keys.")
+            self.weights_mapping_dict = None
+
+    def get_matching_component(self, key: str) -> Optional[str]:
+        """Get the matching component of the key from the weights mapping dictionary. Since the provided mapping
+        can contain partial names of the keys, this function is used to split the key of the current model and
+        match it with the partial key in the mapping, returning the complete name of the key in the pretrained model.
+
+        This allows users to provide one mapping for multiple statistics that share the same prefix. For example,
+        if the mapping is {"model": "global_model"} and the input key of the current model is "model.layer1.weight",
+        then the returned matching component is "global_model.layer1.weight".
+
+        Args:
+            key (str): Key to be matched in pretrained model.
+
+        Returns:
+            Optional[str]: If no weights mapping dict is provided, returns the key. Otherwise, if the key is in the
+            weights mapping dict, returns the matching component of the key. Otherwise, returns None.
+        """
+
+        if self.weights_mapping_dict is None:
+            return key
+
+        components = key.split(".")
+
+        for i, component in enumerate(components):
+            if i == 0:
+                matching_component = components[0]
+            else:
+                matching_component += "." + component
+            if matching_component in self.weights_mapping_dict:
+                return self.weights_mapping_dict[matching_component] + key[len(matching_component) :]
+        return None
+
+    def load_from_pretrained(self, model: torch.nn.Module) -> torch.nn.Module:
+        """Load the pretrained model into the current model.
+
+        Args:
+            model (torch.nn.Module): Current model.
+        """
+
+        assert self.pretrained_model_state is not None
+
+        current_model_state = model.state_dict()
+
+        matching_state = {}
+        for key in current_model_state.keys():
+            original_state = current_model_state[key]
+
+            pretrained_key = self.get_matching_component(key)
+            log(INFO, f"Matching: {key} -> {pretrained_key}")
+            if pretrained_key is not None:
+                if pretrained_key in self.pretrained_model_state.keys():
+                    pretrained_state = self.pretrained_model_state[pretrained_key]
+                    if original_state.size() == pretrained_state.size():
+                        matching_state[key] = pretrained_state
+                        log(INFO, "Succesful stats matching.")
+                    else:
+                        log(INFO, f"Dismatched sizes {original_state.size()} -> {pretrained_state.size()}.")
```

Comment:
Small typo: Dismatched vs. Mismatched.

I would have this log be a WARNING level log, as it means the key was matched but something happened with the state. I would also log that this results in that the `pretrained_state` not being loaded. That is, we skip it so the user knows that it will be the effect.